code/src/components/ToDoList.js

```diff
@@ -0,0 +1,242 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import styled, { keyframes } from "styled-components";
+
+import tasks, { selectFilteredTodos } from "../reducers/tasks";
+import moment from "moment";
+
+const ToDoList = () => {
+  const items = useSelector((store) => selectFilteredTodos(store));
+
+  const dispatch = useDispatch();
+
+  const onToggleTask = (id) => {
+    dispatch(tasks.actions.toggleTask(id));
+  };
+
+  const onClickCompleteAll = () => {
+    dispatch(tasks.actions.completeAllTodos());
+  };
+
+  const onDeleteTask = (id) => {
+    dispatch(tasks.actions.deleteTask(id));
+  };
+
+  return (
+    <ToDoContainer>
+      {items.map((item) => (
+        <div key={item.id}>
+          <TasksContainer>
+            <InfoContainer>
+              <Task>{item.text}</Task>
+              <Date>{moment().format("L")}</Date>
+            </InfoContainer>
+            <IconsContainer>
+              <Label>
+                <Input
```

Comment:
Impressed that you got the checkboxes to work =) I always have issue with those.